core/utils/xml.ts

```diff
@@ -108,10 +111,16 @@ export function createTextNode(text: string): Text {
  * @throws if the text doesn't parse.
  */
 export function textToDom(text: string): Element {
-  const doc = textToDomDocument(text);
+  let doc = textToDomDocument(text);
   if (!doc || !doc.documentElement ||
       doc.getElementsByTagName('parsererror').length) {
-    throw Error('textToDom was unable to parse: ' + text);
+    const oParser = new DOMParser();
+    doc = oParser.parseFromString(text, 'text/html');
+    if (!doc || !doc.body.firstChild ||
+        doc.body.firstChild.nodeName.toLowerCase() != 'xml') {
```

Comment:
Aren't we doing `!==` these days?

Reply:
Whoops, was pulled over from the previous PR